tools/autograd/derivatives.yaml

```diff
@@ -1532,11 +1532,11 @@
   result: auto_linear
 
 - name: sum.SymInt(Tensor self, SymInt[1] dim, bool keepdim=False, *, ScalarType? dtype=None) -> Tensor
-  self: sum_backward(grad, self.sym_sizes(), dim, keepdim)
+  self: sum_backward_symint(grad, self.sym_sizes(), dim, keepdim)
   result: auto_linear
 
 - name: sum.dim_IntList(Tensor self, int[1]? dim, bool keepdim=False, *, ScalarType? dtype=None) -> Tensor
-  self: sum_backward(grad, self.sizes(), dim, keepdim)
+  self: sum_backward(grad, IntArrayRef({-1}), dim, keepdim, self)
```

Comment:
The problem is that this is not OK to do. The reason is that if you use `self` here, then the autograd has to save the full Tensor of size `self` and thus this will increase memory usage for all users.

Reply:
@albanD completely agree that what I'm doing here is horrible, was just hacking together something to try to get it to compile! If I understand correctly from talking to Driss before, (please correct me if I'm wrong) the scheme in the linked google doc was not implemented for now. I think I heard during the meeting yesterday that it was possible to workaround the `self.sizes()` issue for NestedTensor in VariableTypeManual.cpp though, would you be willing to point me to an example of how this could be done?